pkg/cmd/roachtest/tests/backup.go

```diff
@@ -164,58 +171,338 @@ func registerBackupNodeShutdown(r registry.Registry) {
 	r.Add(registry.TestSpec{
-		Name:            "backup/mixed-version",
+		Name:            "backup/mixed-version-basic",
 		Owner:           registry.OwnerBulkIO,
 		Cluster:         r.MakeClusterSpec(4),
 		EncryptAtRandom: true,
+		RequiresLicense: true,
 		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
 			// An empty string means that the cockroach binary specified by flag
 			// `cockroach` will be used.
 			const mainVersion = ""
 			roachNodes := c.All()
+			upgradedNodes := c.Nodes(1, 2)
+			oldNodes := c.Nodes(3, 4)
 			predV, err := PredecessorVersion(*t.BuildVersion())
 			require.NoError(t, err)
 			c.Put(ctx, t.DeprecatedWorkload(), "./workload")
-			loadBackupDataStep := func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-				rows := rows3GiB
-				if c.IsLocal() {
-					rows = 100
-				}
-				runImportBankDataSplit(ctx, rows, 0 /* ranges */, t, u.c)
-			}
-			successfulBackupStep := func(nodeID int, opts ...string) versionStep {
-				return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-					backupOpts := ""
-					if len(opts) > 0 {
-						backupOpts = fmt.Sprintf("WITH %s", strings.Join(opts, ", "))
-					}
-					backupQuery := fmt.Sprintf("BACKUP bank.bank TO 'nodelocal://%d/%s' %s",
-						nodeID, destinationName(c), backupOpts)
-
-					gatewayDB := c.Conn(ctx, t.L(), nodeID)
-					defer gatewayDB.Close()
-					t.Status("Running: ", backupQuery)
-					_, err := gatewayDB.ExecContext(ctx, backupQuery)
-					require.NoError(t, err)
-				}
-			}
 			u := newVersionUpgradeTest(c,
 				uploadAndStartFromCheckpointFixture(roachNodes, predV),
 				waitForUpgradeStep(roachNodes),
 				preventAutoUpgradeStep(1),
-				loadBackupDataStep,
-				// Upgrade some of the nodes.
-				binaryUpgradeStep(c.Nodes(1, 2), mainVersion),
-				// Backup from new node should succeed.
-				successfulBackupStep(1),
-				// Backup from new node with revision history should succeed.
-				successfulBackupStep(2, "revision_history"),
-				// Backup from old node should succeed.
-				successfulBackupStep(3),
-				// Backup from old node with revision history should succeed.
-				successfulBackupStep(4, "revision_history"),
+				setShortJobIntervalsStep(1),
+				loadBackupDataStep(c),
+				// Upgrade some nodes.
+				binaryUpgradeStep(upgradedNodes, mainVersion),
+
+				// Let us first test planning and executing a backup on different node
+				// versions.
+				//
+				// NB: All backups in this test are writing to node 1's ExternalIODir
+				// for simplicity.
+
+				// Case 1: plan backup    -> old node
+				//         execute backup -> upgraded node
+				//
+				// Halt job execution on older nodes.
+				disableJobAdoptionStep(c, oldNodes),
+
+				// Run a backup from an old node so that it is planned on the old node
+				// but the job is adopted on a new node.
+				planAndRunBackup(t, c, oldNodes.RandNode(), oldNodes,
+					`BACKUP TABLE bank.bank INTO 'nodelocal://1/plan-old-resume-new' WITH detached`),
```

Comment:
right now, the test only covers mixed version (planning and execution on different versions) for full backups. What if we ran an incremental backup as well ( i.e. just another `planAndRunBackup` call for each case)? Seems like an easy way to get a bit more coverage!

Reply:
done!